.github/workflows/test.yml

```diff
@@ -23,6 +23,12 @@ jobs:
       - name: Run tests
         run: pnpm test
 
+      - name: Generate gas reports
+        run: pnpm gas-report
```

Comment:
This is moved here since the gas report runs best if the tests are already compiled/cached.